locations/spiders/texas_roadhouse.py

```diff
@@ -0,0 +1,153 @@
+import datetime
+import xml.etree.ElementTree as ET
+
+import scrapy
+
+from locations.items import GeojsonPointItem
+from locations.hours import OpeningHours
+
+MAX_RETRIES = 3
+DAY_MAPPING = {
+    'MonHours': 'Mo',
+    'TueHours': 'Tu',
+    'WedHours': 'We',
+    'ThuHours': 'Th',
+    'FriHours': 'Fr',
+    'SatHours': 'Sa',
+    'SunHours': 'Su'
+}
+
+
+class TexasRoadhouseSpider(scrapy.Spider):
+    name = "texas_roadhouse"
+    allowed_domains = ["www.texasroadhouse.com"]
+    download_delay = 0.5
+    start_urls = (
+        'https://www.texasroadhouse.com/sitefinity/services/LeapFrog/TRH_Proxy.asmx/GetAllStoresDataSet',
+    )
+
+    @staticmethod
+    def format_address(data):
+        addr = " ".join(filter(None, [data["Address1"], data["Address2"]]))
+        city = data["City"]
+        state = data["State"].strip()
+        zip = data["Zip"]
+        country = data["Country"]
+
+        if not state and not zip:  # international address - does not have state/zip but may have city
+            if city:
+                return "{}, {}, {}".format(addr, city, data["Country"])
+            else:
+                return "{}, {}".format(addr, data["Country"])
+        else:
+            return "{}, {}, {} {}, {}".format(addr, city, state, zip, country)
+
+    @staticmethod
+    def convert_to_24hr(date):
+        """Convert to 24hour
+        Eg: 04:00PM --> 16:00
+        """
+        return datetime.datetime.strptime(date, '%I:%M%p').strftime('%H:%M')
+
+    def parse_hours(self, data):
+        """
+        Eg:
+        {
+            "MonHours": "4:00PM-10:00PM",
+            "TueHours": "4:00PM-10:00PM",
+            "WedHours": "4:00PM-10:00PM",
+            "ThuHours": "4:00PM-10:00PM",
+            "FriHours": "4:00PM-11:00PM",
+            "SatHours": "11:00AM-11:00PM",
+            "SunHours": "11:00AM-10:00PM"
+        }
+        :param hours:
+        :return:
+        """
+        hours = OpeningHours()
+        for k, v in DAY_MAPPING.items():
+            open, close = data[k].split('-')
+            if not open or not close:
+                continue
+            hours.add_range(day=v,
+                            open_time=self.convert_to_24hr(open),
+                            close_time=self.convert_to_24hr(close))
+
+        return hours.as_opening_hours()
+
+    def parse_store(self, response):
+        retries = 0
+        ref = response.meta['id']
+        name = response.meta['name']
+
+        try:
+            root = ET.fromstring(response.text)
```

Comment:
The response already comes in parsed as XML. You can use `response.xpath()` to query the XML document so you don't have to re-parse it inside the spider.

Reply:
@iandees  Thanks!  I made the switch to response.xpath and cleaned up some other stuff.